### PR TITLE
feat(release): Add craft release

### PR DESCRIPTION
**Pull request by hubertdeng123**
adds craft release to github, pypi, sentry-pypi

### CODE REVIEW

File: setup.cfg
Comment (hubertdeng123) [inline setup.cfg:1]: apparently flake8 doesn't support using pyproject.toml yet so can't remove setup.cfg just yet

Comment (hubertdeng123) [reply setup.cfg:1]: https://github.com/PyCQA/flake8/issues/234

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      force:
+        description: Force a release even when there are release-blockers (optional)
+        required: false
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    name: "Release a new version"
+    steps:
+      - uses: actions/checkout@v4
```
Comment (IanWoodard) [inline .github/workflows/release.yml:18]:
```suggestion
      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
```

For consistency and to pin it.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        with:
+          token: ${{ secrets.GH_RELEASE_PAT }}
+          fetch-depth: 0
+      - name: Prepare release
+        uses: getsentry/action-prepare-release@v1.6.4
```
Comment (IanWoodard) [inline .github/workflows/release.yml:23]:
```suggestion
        uses: getsentry/action-prepare-release@3cea80dc3938c0baf5ec4ce752ecb311f8780cdc # v1.6.4
```
For consistency and to pin it.

**Review by IanWoodard — APPROVED** [review]

**Pull request merged**